doc/source/whatsnew/v0.21.0.txt

```diff
@@ -201,7 +201,8 @@ I/O
 ^^^
 
 - Bug in :func:`read_csv` in which non integer values for the header argument generated an unhelpful / unrelated error message (:issue:`16338`)
-
+- Bug in :func:`read_csv` in which passing a CSV with at least one very large (i.e. more than 2^31 - 1 bytes) column along with ``low_memory=False`` would cause an integer overflow. The result was an always unsuccessful attempt to allocate an enourmous buffer and then reporting "Out of memory." (:issue:`16798`).
+- Bug in :func:`read_csv` in which some errors paths were assigning error messages to the internal tokenizer's ``error_msg`` field without first allocating the memory. When this happened as part of exception handling, it resulted in a double ``free`` and the program halted due to a ``SIGSEGV`` (:issue:`16798`).
```

Comment:
I wonder if we can make these descriptions less technical.  Our (less technically) audience might not understand everything written here.  For example, you could write:
~~~
- Bug in :func:`read_csv` in which passing a CSV with at least one very large...would cause Python to run out of memory.
- Bug in :func:`read_csv` in which memory mismanagement with error messages caused Python to crash
~~~

True, it doesn't capture the entire error description as before, but it gets the general idea across I hope.